### PR TITLE
feat: [PL-38509]: Added apiKey resource

### PR DESCRIPTION
**Pull request by VikasMaddukuriHarness**
## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`

### CODE REVIEW

File: internal/provider/provider.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ import (
 	cd_trigger "github.com/harness/terraform-provider-harness/internal/service/cd/trigger"
 	"github.com/harness/terraform-provider-harness/internal/service/cd/user"
 	"github.com/harness/terraform-provider-harness/internal/service/cd/yamlconfig"
+	pl_apiKey "github.com/harness/terraform-provider-harness/internal/service/platform/api_key"
```
Comment (rijajoo) [inline internal/provider/provider.go:35]: pl_api_key (Casing)

File: internal/service/platform/api_key/apiKey.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+		Importer:      helpers.MultiLevelResourceImporter,
+
+		Schema: map[string]*schema.Schema{
+			"identifier": {
+				Description: "Unique identifier of the resource",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"name": {
+				Description: "Name of the ApiKey",
```
Comment (rijajoo) [inline internal/service/platform/api_key/apiKey.go:33]: Name of the Api Key

File: internal/service/platform/api_key/data_source_apiKey.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+			return helpers.HandleApiError(err, d, httpResp)
+		}
+
+		if apiKey == nil {
+			d.SetId("")
+			d.MarkNewResource()
+			return nil
+		}
+	} else {
+		return diag.FromErr(errors.New("identifier  must be specified"))
```
Comment (rijajoo) [inline internal/service/platform/api_key/data_source_apiKey.go:108]: Identifier must be specified (spacing, capitalization)

**Review by rijajoo — CHANGES_REQUESTED** [review]
Please address all these comments in all places. Also please add changelog for newer resource added.

**Review by rijajoo — CHANGES_REQUESTED** [review]

**Comment by rathodmeetsatish** [comment]
trigger build

**Comment by adiyaar24** [comment]
trigger subcategoryfieldcheck

**Review by adiyaar24 — APPROVED** [review]

**Pull request merged**